core/classifiers/TextClassifier/TextClassifier.py

```diff
@@ -0,0 +1,170 @@
+        logging.info(
+            'train() spent %f seconds for %d instances' % (end - start,
+                                                           len(x)))
+        self.best_clf = clf.best_estimator_
+        self.best_params = clf.best_params_
+        self.best_score = clf.best_score_
+        self.count_vector = count_vector
+        self.exec_time = end-start
+
+    def to_dict(self):
+        """Returns a dict representing this classifier.
+
+        Returns:
+            dict. A dictionary representation of classifier referred to
+            as 'classifier_data'. This data is used for prediction.
+        """
+        classifier_data = {
+            'SVM': classifier_utils.extract_svm_parameters(self.best_clf),
+            'cv_vocabulary': self.count_vector.__dict__['vocabulary_'],
+            'best_params': self.best_params,
+            'best_score': self.best_score
+        }
+        return classifier_data
+
+    def validate(self, classifier_data):
+        """Validates classifier data.
+
+        Args:
+            classifier_data: dict of the classifier attributes specific to
+                the classifier algorithm used.
+        """
+        allowed_top_level_keys = ['SVM', 'cv_vocabulary', 'best_params',
+                                  'best_score']
+        for key in allowed_top_level_keys:
+            if key not in classifier_data:
+                raise Exception(
+                    '\'%s\' key not found in classifier_data.' % key)
+
+            if key != 'best_score':
+                if not isinstance(classifier_data[key], dict):
+                    raise Exception(
+                        'Expected  \'%s\' to be dict but found \'%s\'.'
+                        % (key, type(classifier_data[key])))
+            else:
+                if not isinstance(classifier_data[key], float):
+                    raise Exception(
+                        'Expected  \'%s\' to be float but found \'%s\'.'
+                        % (key, type(classifier_data[key])))
+
+        allowed_best_params_keys = ['kernel', 'C']
+        for key in allowed_best_params_keys:
+            if key not in classifier_data['best_params']:
+                raise Exception(
+                    '\'%s\' key not found in \'best_params\''
+                    ' in classifier_data.' % key)
+
+        if not isinstance(classifier_data['best_params']['C'], float):
+            raise Exception(
+                'Expected \'C\' to be a float but found \'%s\'' %
+                type(classifier_data['best_params']['C']))
+
+        if not isinstance(classifier_data['best_params']['kernel'],
+                          basestring):
+            raise Exception(
+                'Expected \'kernel\' to be a string but found \'%s\'' %
+                type(classifier_data['best_params']['kernel']))
+
+        # Validate that entire classifier data is json serializable and
+        # does not raise any exception.
+        json.dumps(classifier_data)
```

Comment:
we can do that but keys like "SVM" of classifier_data are specific to the algorithm being used. In that case, we would be checking for `allowed_top_level_keys` twice - once in the validate() of BaseClassifier and a second time in the validate() of derived class. Why not just let the implementer decide what they want to put in the classifier data and rather enforce the implementation of validate() method which they can implement as per their need?